components/style/values/computed/transform.rs

```diff
@@ -25,3 +34,139 @@ impl TransformOrigin {
+                                  -> Option<Transform3D<CSSFloat>> {
+        let mut transform = Transform3D::identity();
+        let list = match self.0.as_ref() {
+            Some(list) => list,
+            None => return None,
+        };
+
+        let extract_pixel_length = |lop: &LengthOrPercentage| {
+            match *lop {
+                LengthOrPercentage::Length(au) => au.to_f32_px(),
+                LengthOrPercentage::Percentage(_) => 0.,
+                LengthOrPercentage::Calc(calc) => calc.length().to_f32_px(),
+            }
+        };
+
+        for operation in list {
+            let matrix = match *operation {
+                ComputedOperation::Rotate(ax, ay, az, theta) => {
+                    let theta = Angle::from_radians(2.0f32 * f32::consts::PI - theta.radians());
+                    let (ax, ay, az, theta) =
+                        Self::get_normalized_vector_and_angle(ax, ay, az, theta);
+                    Transform3D::create_rotation(ax, ay, az, theta.into())
+                }
+                ComputedOperation::Perspective(d) => {
+                    Self::create_perspective_matrix(d)
+                }
+                ComputedOperation::Scale(sx, sy, sz) => {
+                    Transform3D::create_scale(sx, sy, sz)
+                }
+                ComputedOperation::Translate(tx, ty, tz) => {
+                    let (tx, ty) = match reference_box {
+                        Some(relative_border_box) => {
+                            (tx.to_used_value(relative_border_box.size.width).to_f32_px(),
+                             ty.to_used_value(relative_border_box.size.height).to_f32_px())
+                        },
+                        None => {
+                            // If we don't have reference box, we cannot resolve the used value,
+                            // so only retrieve the length part. This will be used for computing
+                            // distance without any layout info.
+                            (extract_pixel_length(&tx), extract_pixel_length(&ty))
+                        }
+                    };
+                    let tz = tz.to_f32_px();
+                    Transform3D::create_translation(tx, ty, tz)
+                }
+                ComputedOperation::Matrix(m) => {
+                    m.into()
+                }
+                ComputedOperation::MatrixWithPercents(_) => {
+                    // `-moz-transform` is not implemented in Servo yet.
+                    unreachable!()
+                }
+                ComputedOperation::Skew(theta_x, theta_y) => {
+                    Transform3D::create_skew(theta_x.into(), theta_y.into())
+                }
+                ComputedOperation::InterpolateMatrix { .. } |
+                ComputedOperation::AccumulateMatrix { .. } => {
+                    // TODO: Convert InterpolateMatrix/AccmulateMatrix into a valid Transform3D by
```

Comment:
nit: There's a typo here (`AccmulateMatrix`)

Reply:
I will update this comment in other PR. Thanks for this!